src/modules/iasql_functions/iasql.ts

```diff
@@ -783,91 +798,262 @@ async function getChangeLogsSinceLastBegin(orm: TypeormWrapper): Promise<IasqlAu
+            .filter(([_, v]: [string, any]) => v !== null)
+            .map(async ([k, v]: [string, any]) => {
+              return await augmentValue(
+                cl.tableName,
+                k,
+                v,
+                modsIndexedByTable,
+                orm,
+                shouldRecreateSubQueries,
+                recreatedEntitiesFromChangelogs,
+              );
+            }),
         );
-        changedAugmentedValues = Object.entries(cl.change.change ?? {})
-          .map(([k, v]: [string, any]) => augmentValue(k, v, tableMetadata))
-          .filter(av => (!av.isPrimary && !av.isAutogenerated) || !av.isAutogenerated);
-        inverseQuery = format(
-          `
-            UPDATE %I
-            SET ${originalAugmentedValues.map(() => `%I = %s`).join(', ')}
-            WHERE ${changedAugmentedValues
-              // We need to add an special case for AMIs since we know the revolve string can be used and it will not match with the actual AMI assigned
-              .filter(av => av.key !== 'ami')
-              .map(
-                av =>
-                  `${av.formattedValue === 'NULL' ? '%I IS %s' : av.isJson ? '%I::jsonb = %s' : '%I = %s'}`,
-              )
-              .join(' AND ')};
-          `,
+        const augmentedChangedEntries = await Promise.all(
+          Object.entries(cl.change.change ?? {})
+            .filter(([_, v]: [string, any]) => v !== null)
+            .map(async ([k, v]: [string, any]) => {
+              return await augmentValue(
+                cl.tableName,
+                k,
+                v,
+                modsIndexedByTable,
+                orm,
+                shouldRecreateSubQueries,
+                recreatedEntitiesFromChangelogs,
+              );
+            }),
+        );
+        // Get formatted query
+        query = await getFormattedQuery(
           cl.tableName,
-          ...originalAugmentedValues.flatMap(av => [av.key, av.formattedValue]),
-          ...changedAugmentedValues
-            // We need to add an special case for AMIs since we know the revolve string can be used and it will not match with the actual AMI assigned
-            .filter(av => av.key !== 'ami')
-            .flatMap(av => [av.key, av.formattedValue]),
+          AuditLogChangeType.UPDATE,
+          inverse,
+          augmentedOriginalEntries,
+          augmentedChangedEntries,
         );
         break;
+      }
       default:
         break;
     }
-    if (inverseQuery) inverseQueries.push(inverseQuery);
+    if (query) queries.push(query);
+  }
+  return queries;
+}
+
+/**
+ * @internal
+ * Recreate entities from change logs indexed by entity name
+ */
+async function recreateEntitiesFromChangelogs(
+  changeLogs: IasqlAuditLog[],
+  modsIndexedByTable: ModsIndexedByTable,
+  orm: TypeormWrapper,
+): Promise<EntitiesIndexesByEntityName> {
+  const entityMapper: { [key: string]: MapperBase<any> } = {};
```

Comment:
Nit: This logically should be within the `for` loop, as far as I can tell. It's never used outside of the for loop